projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -507,6 +508,10 @@ public static function queue_historically_active_jetpack_modules_update() {
 	 * @return void
 	 */
 	public static function setup_historically_active_jetpack_modules_sync() {
+		if ( wp_doing_ajax() ) {
+			return;
+		}
+
```

Comment:
This is probably not needed here - since this function is (mostly) just listening in for other actions and deferring the actual update of the plugins array for the next page load.

You might move the check into the following conditional statement so that an update of the historically active plugins array does not happen during an ajax call:

```
if ( get_transient( self::UPDATE_HISTORICALLY_ACTIVE_JETPACK_MODULES_KEY )  &&  ! wp_doing_ajax() ) {

```

Also, the `activated_plugin` action could very likely be happening in an async call.

Reply:
Good call, thank you